### PR TITLE
Build package on PR

### PR DESCRIPTION
**Pull request by ikifar2012**
This PR will allow us to automatically generate a Debian package when someone makes a PR thus making it easier to test 
image

For more info read:
https://github.com/actions/upload-artifact#where-does-the-upload-go

Github Actions seems to have a storage limit per month 
image
Source:  https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions

Thus I have implemented a max age of 15 days after which the package is deleted (although each package is only around 4.33 KB)

### CODE REVIEW

**Comment by ikifar2012** [comment]
Just want to make sure before merging this, that we won't go over any limits, @pvizeli can you confirm that this is ok?

**Comment by pvizeli** [comment]
And you can download the artifacts after it is inside the browser?

File: .github/workflows/build-deb-pr.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Build Test Deb Package
+
+# yamllint disable-line rule:truthy
+on:
+  workflow_dispatch:
```
Comment (ludeeus) [inline .github/workflows/build-deb-pr.yml:5]: It's better to have this build on push to the default branch instead of using `workflow_dispatch`

File: .github/workflows/build-deb-pr.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  pull_request:
```
Comment (ludeeus) [inline .github/workflows/build-deb-pr.yml:6]: limit this to pull requests opened to the default branch

File: .github/workflows/build-deb-pr.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+      - run: chmod 755 homeassistant-supervised/DEBIAN/p*
+      - run: dpkg-deb --build --root-owner-group homeassistant-supervised
+      - name: Upload
+        uses: actions/upload-artifact@v2
+        with:
+          name: homeassistant-supervised.deb
+          path: homeassistant-supervised.deb
+          retention-days: 15
```
Comment (ludeeus) [inline .github/workflows/build-deb-pr.yml:20]:
```suggestion
```

These files are small, and this repo does not have that high of a volume of builds, the default of 90 is fine

File: .github/workflows/build-deb-pr.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+on:
+  workflow_dispatch:
+  pull_request:
+jobs:
+  build-test-deb-package:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - run: chmod 755 homeassistant-supervised/DEBIAN/p*
```
Comment (ludeeus) [inline .github/workflows/build-deb-pr.yml:13]: Why is this needed?
Is the git index not correct?
755 seems overkill?

File: .github/workflows/build-deb-pr.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      - run: dpkg-deb --build --root-owner-group homeassistant-supervised
```
Comment (ludeeus) [inline .github/workflows/build-deb-pr.yml:14]: Is the `--root-owner-group` really needed here?

File: .github/workflows/build-deb-pr.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+name: Build Test Deb Package
+
+# yamllint disable-line rule:truthy
```
Comment (ludeeus) [inline .github/workflows/build-deb-pr.yml:3]:
```suggestion
```

We don't run YAML lint, no need to exclude this line for something that does not run.

**Review by ludeeus — CHANGES_REQUESTED** [review]

**Comment by ikifar2012** [comment]
> And you can download the artifacts after it is inside the browser?

yes

File: .github/workflows/build-deb-pr.yml (hunk 5 above)
Comment (ikifar2012) [reply .github/workflows/build-deb-pr.yml:13]: It seems it is, read step 5 here https://www.internalpointers.com/post/build-binary-deb-package-practical-guide

File: .github/workflows/build-deb-pr.yml (hunk 4 above)
Comment (ikifar2012) [reply .github/workflows/build-deb-pr.yml]: I have set it to 555 now

**Review by ludeeus — APPROVED** [review]
LGTM
